File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let sum = 0;
    let maxi = -Infinity;
    for(let i=0; i<nums.length; i++){
        sum += nums[i];
        
        if(maxi < sum)
            maxi = sum;
        
        if(sum < 0)
            sum = 0;
    }
    return maxi;
};

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxi = INT_MIN, sum = 0;
        for(auto it: nums){
            sum += it;
            
            if(maxi < sum)
                maxi = sum;
            
            if(sum < 0) 
                sum = 0;
            
        }
        return maxi;
    }
};